packages/data-point-codemods/transforms/hash-and-collection-execution-order.js

```diff
@@ -0,0 +1,65 @@
+const partition = require('lodash/partition')
+const findLastIndex = require('lodash/findLastIndex')
+
+const { getEntityObjects } = require('./utils')
+
+function isMatchingKey (key, keys) {
+  const id = key.type === 'Identifier' ? key.name : key.value
+  return keys.includes(id)
+}
+
+module.exports = (file, api, options) => {
+  const j = api.jscodeshift
+  const root = j(file.source)
+
+  function updateEntity (node, keys) {
+    let [props, otherProps] = partition(node.value.properties, prop => {
+      return isMatchingKey(prop.key, keys)
+    })
+
+    if (props.length < 2) {
+      return
+    }
+
+    props = keys.reduce((acc, key) => {
+      const prop = props.find(p => isMatchingKey(p.key, [key]))
+      if (prop) {
+        acc.push(prop)
+      }
+
+      return acc
+    }, [])
+
+    const composeProp = j.property(
+      'init',
+      j.identifier('compose'),
+      j.arrayExpression(
+        props.map(prop => {
+          return j.objectExpression([prop])
+        })
+      )
+    )
+
+    const index = findLastIndex(otherProps, prop => {
+      return isMatchingKey(prop.key, ['inputType', 'before', 'value'])
+    })
+
+    if (index === -1) {
+      node.value.properties = [composeProp].concat(otherProps)
+    } else {
+      otherProps.splice(index + 1, 0, composeProp)
+      node.value.properties = otherProps
+    }
+  }
+
+  function execute (entityId, keys) {
+    const nodes = getEntityObjects(entityId, j, root)
+    nodes[0].forEach(node => updateEntity(node, keys))
+    nodes[1].forEach(node => updateEntity(node, keys))
+  }
+
+  execute('collection', ['filter', 'map', 'find'])
+  execute('hash', ['omitKeys', 'pickKeys', 'mapKeys', 'addValues', 'addKeys'])
+
+  return root.toSource()
```

Comment:
In the codemod for changing access to the acc root we have a means of checking the quote style to preserve it. Do you think it's worth adding that here? If so, it might be worth pulling the function that's used into a utility while we're at it.

https://github.com/ViacomInc/data-point/blob/756ba053b9c6f7d77524a981b5b5893d4cd57bf1/packages/data-point-codemods/transforms/change-path-reducer-accessing-root-path.js#L55-L61